chain/x/zoracle/exec_env.go

```diff
@@ -63,20 +116,19 @@ func (env *ExecutionEnvironment) GetValidatorAddress(validatorIndex int64) ([]by
 	return env.request.RequestedValidators[validatorIndex], nil
 }
 
-func (env *ExecutionEnvironment) GetMaximumResultSize() int64 {
-	return int64(env.keeper.GetParam(env.ctx, KeyMaxResultSize))
-}
-
-func (env *ExecutionEnvironment) GetMaximumCalldataOfDataSourceSize() int64 {
-	return int64(env.keeper.GetParam(env.ctx, KeyMaxCalldataSize))
-}
-
 func (env *ExecutionEnvironment) RequestExternalData(
 	dataSourceID int64,
 	externalDataID int64,
 	calldata []byte,
-) error {
-	return env.keeper.AddNewRawDataRequest(env.ctx, env.requestID, types.ExternalID(externalDataID), types.DataSourceID(dataSourceID), calldata)
+) {
+	// We can safely skip appending rawDataRequests if its size already surpasses maxRawDataRequestCount threshold since it will fail the transaction regardless.
+	if int64(len(calldata)) <= env.maxCalldataSize && int64(len(env.rawDataRequests)) <= env.maxRawDataRequestCount {
```

Comment:
if `int64(len(calldata)) > env.maxCalldataSize` happens, we should just return -1 back to the requester, I think. Same for when `int64(len(env.rawDataRequests)) >= env.maxRawDataRequestCount`